File: helloworld-server/userserver/src/main/java/com/example/helloworld/userserver/member/dto/response/MemberProfileResponse.java
package com.example.helloworld.userserver.member.dto.response;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import io.swagger.v3.oas.annotations.media.Schema;

import java.sql.Timestamp;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Schema(name = "MemberProfileResponse")
public record MemberProfileResponse(
        @Schema(description = "회원 프로필")
        MemberBlock member
) {
    @Schema(name = "MemberBlock")
    public record MemberBlock(
            Long id,
            @JsonProperty("google_email") String googleEmail,
            String nickname,
            String gender,               // "female" | "male"
            Integer age,
            @JsonProperty("image_url") String imageUrl
    ) {}

    @JsonInclude(JsonInclude.Include.NON_NULL) // 부분 수정용
    public static record MemberUpdateRequest(
            String nickname,
            Integer age
    ) {}

    }

File: helloworld-android/app/src/main/java/com/ms/helloworld/dto/response/MemberProfileResponse.kt
package com.ms.helloworld.dto.response

import com.google.gson.annotations.SerializedName

data class MemberProfileResponse(
    val member: MemberProfile,
    val couple: CoupleProfile?
)

data class MemberProfile(
    val id: Long,
    @SerializedName("google_email") val googleEmail: String?,
    val nickname: String?,
    val gender: String?, // "FEMALE" or "MALE"
    @SerializedName("image_url") val imageUrl: String?,
    val age: Int?
)

data class CoupleProfile(
    @SerializedName("couple_id") val coupleId: Long,
    @SerializedName("user_a_id") val userAId: Long?,
    @SerializedName("user_b_id") val userBId: Long?,
    @SerializedName("pregnancyWeek") val pregnancyWeek: Int?,  // 서버가 camelCase로 보냄
    @SerializedName("due_date") val dueDate: String?, // "yyyy-MM-dd" format
    @SerializedName("menstrual_date") val menstrualDate: String?, // "yyyy-MM-dd" format
    @SerializedName("is_childbirth") val isChildbirth: Boolean?
)

data class PartnerInfo(
    val id: Long,
    val nickname: String?,
    val gender: String?,
    val imageUrl: String?
)